src/main/java/com/netgrif/application/engine/integration/plugins/service/PluginService.java

```diff
@@ -0,0 +1,115 @@
+package com.netgrif.application.engine.integration.plugins.service;
+
+import com.google.protobuf.ByteString;
+import com.netgrif.application.engine.integration.plugin.injector.PluginInjector;
+import com.netgrif.application.engine.integration.plugins.domain.Plugin;
+import com.netgrif.application.engine.integration.plugins.properties.PluginRegistrationConfigProperties;
+import com.netgrif.application.engine.integration.plugins.repository.PluginRepository;
+import com.netgrif.pluginlibrary.core.*;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.SerializationUtils;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Base service that manages gRPC server on application startup, registers, activates and deactivates plugins, sends
+ * plugin execution requests to desired plugin.
+ * */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class PluginService implements IPluginService {
+    private final PluginRepository pluginRepository;
+    private final PluginRegistrationConfigProperties properties;
+    private Server server;
+
+    @PostConstruct
+    public void startServer() throws IOException {
+        server = ServerBuilder
+                .forPort(properties.getPort())
+                .addService(new PluginRegistrationService(this))
+                .build();
+        server.start();
+        log.info("[gRPC Server] - Started on port " + properties.getPort());
+    }
+
+    @PreDestroy
+    public void stopServer() {
+        server.shutdown();
+        log.info("[gRPC Server] - Sopped server on port " + properties.getPort());
+    }
+
+    /**
+     * @param plugin - plugin to be registered, or if already registered, then activate
+     * */
+    @Override
+    public void register(Plugin plugin) {
+        Plugin existingPlugin = pluginRepository.findByIdentifier(plugin.getIdentifier());
+        if (existingPlugin != null) {
+            log.warn("Plugin with identifier \"" + plugin.getIdentifier() + "\" has already been registered. Plugin will be activated.");
```

Comment:
I would prefer separate functions for registering, unregistering, activating, and deactivation plugin